components/widget/filter_buttons/widget_filter_buttons.lua

```diff
@@ -0,0 +1,121 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/FilterButtons
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local FnUtil = require('Module:FnUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Variables = require('Module:Variables')
+local Table = require('Module:Table')
+
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Div = HtmlWidgets.Div
+local FilterButton = Lua.import('Module:Widget/FilterButtons/Button')
+local FilterButtonRow = Lua.import('Module:Widget/FilterButtons/ButtonRow')
+local Widget = Lua.import('Module:Widget')
+
+---@class FilterButtons: Widget
+---@operator call(table): FilterButtons
+local FilterButtons = Class.new(Widget)
+
+---@class FilterButtonCategory
+---@field name string
+---@field property string
+---@field items string[]?
+---@field defaultItems string[]?
+---@field defaultItem string?
+---@field itemToPropertyValues? fun(item: string): string?
+---@field itemIsValid? fun(item: string): boolean
+---@field transform? fun(item: string): string?
+---@field expandKey string?
+---@field expandable boolean?
+---@field order? fun(a: string, b: string): boolean
+---@field load? fun(cat: FilterButtonCategory)
+---@field hasFeatured boolean?
+---@field featuredByDefault boolean?
+
+---@return Widget
+function FilterButtons:render()
+	---@type FilterButtonCategory[]
+	local categories = self.props.categories or Lua.import('Module:FilterButtons/Config').categories
+
+	Array.forEach(categories, FilterButtons._loadCategories)
+
+	return Div{
+		children = Array.map(categories, function (category)
+			-- Variable used to pass default values on to other modules using these.
```

Comment:
Comment should be moved directly to the varDefine
(Not even sure if anything uses the var anymore besides my test module, iirc the new modules directly load the Config to get default items, or don't even check for defaults. Removal OOS)

Reply:
fwiw this is 1-1 from the current module ;)
--> [96b21da](https://github.com/Liquipedia/Lua-Modules/pull/5547/commits/96b21dafc7a7e541af48cdd8a42d7cfe5e676c20)
(it is basically just a move and adjusting the call a tiny bit so it is a widget

i just had to re-create the old module to call the new one so we do not break the current on wiki usage (in wiki code) and so i could test it properly
--> [377f6a6](https://github.com/Liquipedia/Lua-Modules/pull/5547/commits/377f6a6b201b514b241dc3ad1eb583921f0a910b)

Reply:
Fair, i missed that this isn't a merge to main yet.
Will check whether it's used and potentially remove the var when i have time.